include/rapidjson/reader.h

```diff
@@ -1483,14 +1484,90 @@ class GenericReader {
         // Parse minus
         bool minus = Consume(s, '-');
 
-        // Parse int: zero / ( digit1-9 *DIGIT )
+        // Parse int: zero / hexadecimal / ( digit1-9 *DIGIT )
         unsigned i = 0;
         uint64_t i64 = 0;
         bool use64bit = false;
+        bool hex = false;
         int significandDigit = 0;
+        // Parse hxadecimal
         if (RAPIDJSON_UNLIKELY(s.Peek() == '0')) {
-            i = 0;
             s.TakePush();
+            unsigned peekVal = s.Peek();
+            if (peekVal == 'x' || peekVal == 'X') {
```

Comment:
This should be within `kParseHexadecimalsFlag`, so that without `kParseHexadecimalsFlag` there should be no difference in performance.

Reply:
Fixed.